app/models/entitlement_search.rb

```diff
@@ -0,0 +1,31 @@
+class EntitlementSearch
+  def initialize(params)
+    @params = params
+  end
+
+  def dependency_status
+    search_params[:dependency_status]
+  end
+
+  def move_type
+    search_params[:move_type]
+  end
+
+  def rank
+    search_params[:rank]
+  end
+
+  def result
+    Entitlement.find_by(rank: rank)
+  end
+
+  def valid?
+    search_params.permitted? && dependency_status.present? && move_type.present? && rank.present?
+  end
+
+  private
+
+  def search_params
+    @search_params ||= @params.permit(:dependency_status, :move_type, :rank)
```

Comment:
[Strong Parameters](http://guides.rubyonrails.org/action_controller_overview.html#strong-parameters) are more of a controller concern than a model concern, but… they work here… 🤷‍♂️